### DISCUSSION TITLE
HomeAssistant-FusionSolar with Huawei EMMA-A02

### DISCUSSION THREAD

**Discussion by violence88** [tijsverkoyen/HomeAssistant-FusionSolar#177] [Ideas]
Hello everyone,

Currently, there is no suitable solution to integrate EMMA-A02 with MODBUS TCP. Therefore, it would be very beneficial to receive all information via the Northbound API. I attempted this with the current version, but unfortunately, I was not successful.

The following was not available:
The first thing to configure is the electricity grid:

- Grid consumption is provided by `sensor.xxx_reverse_active_energy`.
- Return to the grid is given by `sensor.xxx_active_energy_forward_active_energy`.
- For solar panels, the sensor is `sensor.xxx_total_current_day_energy`.

**Comment by theviper2909** [comment]
EMMA-A02 is devTypeId: 23070  which is missing in the code. 
I tried this via python code myself. The API gave me these KPIs:

'active_cap': 255.02, 'ex_active_power': 'N/A', 'power_factor': -1.0, 'ex_reverse_active_cap': 'N/A', 'a_i': 11.0, 'ex_active_cap': 'N/A', 'c_i': 10.9, 'b_i': 10.8, 'a_u': 237.1, 'reactive_power': 0.149, 'c_u': 237.6, 'b_u': 239.4, 'bc_u': 413.0, 'ex_a_i': 0.0, 'ex_c_i': 0.0, 'ex_active_power_c': 0.0, 'ex_b_i': 0.0, 'reverse_active_cap': 1348.1, 'ex_power_factor': 0.0, 'active_power_b': -2.577, 'active_power_a': -2.59, 'ex_active_power_a': 0.0, 'ex_active_power_b': 0.0, 'run_state': 1, 'ex_reactive_power': 'N/A', 'active_power': -7.755, 'ab_u': 412.6, 'ca_u': 410.9, 'active_power_c': -2.588

Would be great if this could be integrated in the code.

Wallbox with 'devName': 'Wallbox', 'devTypeId': 45577 is also not integrated. Could that be integrated as well?
